pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/patch/DeletePatchTest.java

```diff
@@ -0,0 +1,154 @@
+        Patch tissueDeletePatch = new Gson().fromJson(tissueDeleteJson, Patch.class);
+        patcher = PatchFactory.makePatch(tissueDeletePatch, null);
+        Assert.assertTrue(patcher instanceof DeleteTissuePatch);
+    }
+
+    @Test
+    public void testDeleteOncHistoryAndUniquePatch() {
+        String institutionId = testParticipantUtil.createInstitutionForParticipant(participantId);
+        String medicalRecordId = testParticipantUtil.createMRForParticipant(institutionId);
+        String oncHistoryId = testParticipantUtil.createOncHistoryForParticipant(medicalRecordId);
+        createdOncHistoryIds.add(oncHistoryId);
+        String tissueId = testParticipantUtil.createTissueForParticipant(oncHistoryId);
+        createdTissueIds.add(tissueId);
+        String smIdPk = testParticipantUtil.createSmIdForParticipant(tissueId, "USS_VALUE");
+        createdSmIdPks.add(smIdPk);
+        List<String> participantIds = new ArrayList<>();
+        participantIds.add(participantGuid);
+        OncHistoryDetail oncHistoryDetail = testParticipantUtil.getOncHistoryDetailById(oncHistoryId);
+        Assert.assertFalse(oncHistoryDetail.isDeleted());
+        Tissue tissue = oncHistoryDetail.getTissues().stream().findFirst().get();
+        Assert.assertEquals(String.valueOf(tissue.getTissueId()), tissueId);
+        SmId tissueSmId = tissue.getUssSMID().stream().filter(smId -> String.valueOf(smId.getSmIdPk()).equals(smIdPk)).findAny().get();
+        Assert.assertEquals(String.valueOf(tissueSmId.getTissueId()), tissueId);
+
+        String tissueSkIdUniquePatch = "{\"id\":" + tissueId + ",\"user\":\"" + testParticipantUtil.userEmail +
+                "\",\"nameValue\":{\"name\":\"t.skId\",\"value\":\"123\"},\"ddpParticipantId\":\""
+                + participantGuid + "\",\"parent\":\"oncHistoryDetailId\",\"parentId\":" + oncHistoryId +
+                ",\"tableAlias\":\"t\",\"isUnique\":true,\"realm\":\"" + instanceName + "\"}";
+        Patch patch = new Gson().fromJson(tissueSkIdUniquePatch, Patch.class);
+        BasePatch patcher = PatchFactory.makePatch(patch, null);
+        try {
+            patcher.doPatch();
+        } catch (Exception e){
+            log.error("", e);
+        }
+
+        tissueSkIdUniquePatch = "{\"id\":" + tissueId + ",\"user\":\"" + testParticipantUtil.userEmail +
+                "\",\"nameValue\":{\"name\":\"t.skId\",\"value\":\"123\"},\"ddpParticipantId\":\""
+                + participantGuid + "\",\"parent\":\"oncHistoryDetailId\",\"parentId\":" + oncHistoryId +
+                ",\"tableAlias\":\"t\",\"isUnique\":true,\"realm\":\"" + instanceName + "\"}";
+        patch = new Gson().fromJson(tissueSkIdUniquePatch, Patch.class);
+        patcher = PatchFactory.makePatch(patch, null);
+        try {
+            patcher.doPatch();
```

Comment:
If you are expecting an exception here, then you should put an `Assert.fail` after the call to `doPatch`. Otherwise you will not know if this is actually working.

Reply:
Thanks that's a great idea!